File: 0.template/src/service_cpp_pkg/src/acc_client.cpp
#include "rclcpp/rclcpp.hpp"

class AccClientNode : public rclcpp::Node
{
public:
    AccClientNode() : Node("acc_client_node")
    {
    }
};

int main(int argc, char *argv[])
{
    // Initialize the ROS 2 client library
    rclcpp::init(argc, argv);

    // Create a custom node
    auto node = std::make_shared<AccClientNode>();
    // Keep the node alive until it is shut down
    rclcpp::spin(node);
    // Destroy the node
    node.reset();

    // Shutdown the ROS 2 client library
    rclcpp::shutdown();
    return 0;
}

File: 6.actions/src/action_py_pkg/action_py_pkg/acc_client.py
#!/usr/bin/env python3
import rclpy
from rclpy.node import Node
from rclpy.action import ActionClient


class AccClient(Node):
    """A simple ROS2 node that creates an action client for the action "accumulate"."""

    def __init__(self):
        super().__init__("acc_client_node")
        self.get_logger().info("Accumulate Action Client Node has been started!")
        # Create an action client


def main(args=None):
    # Initialize the ROS2 Python client library
    rclpy.init(args=args)

    # Create a ROS2 node, then spin the node to keep it alive, finally destroy the node
    node = AccClient()
    rclpy.spin(node)
    node.destroy_node()

    # Shutdown the ROS2 Python client library
    rclpy.shutdown()


if __name__ == "__main__":
    main()
